packages/cli/src/commands/pg/promote.ts

```diff
@@ -0,0 +1,176 @@
+      body: {
+        name: 'DATABASE',
+        app: {name: app},
+        addon: {name: attachment.addon.name},
+        namespace: attachment.namespace || null,
+        confirm: app,
+      },
+    })
+    ux.action.stop()
+    const currentPooler = attachments.find(a => a.namespace === 'connection-pooling:default' && a.addon?.id === current.addon?.id && a.name === 'DATABASE_CONNECTION_POOL')
+    if (currentPooler) {
+      ux.action.start('Reattaching pooler to new leader')
+      await this.heroku.post('/addon-attachments', {
+        body: {
+          name: currentPooler.name,
+          app: {name: app},
+          addon: {name: attachment.addon.name},
+          namespace: 'connection-pooling:default',
+          confirm: app,
+        },
+      })
+      ux.action.stop()
+    }
+
+    const {body: promotedDatabaseDetails} = await this.heroku.get<PgDatabase>(`/client/v11/databases/${attachment.addon.id}`, {
+      hostname: pgHost(),
+    })
+    if (promotedDatabaseDetails.following) {
+      const unfollowLeaderCmd = `heroku pg:unfollow ${attachment.addon.name}`
+      ux.warn(heredoc(`
+        Your database has been promoted but it is currently a follower database in read-only mode.
+        
+        Promoting a database with ${color.cmd('heroku pg:promote')} doesn't automatically unfollow its leader.
+        
+        Use ${color.cyan(unfollowLeaderCmd)} to stop this follower from replicating from its leader (${color.yellow(promotedDatabaseDetails.leader as string)}) and convert it into a writable database.
+      `))
+    }
+
+    const {body: formation} = await this.heroku.get<Heroku.Formation[]>(`/apps/${app}/formation`)
+    const releasePhase = formation.find(process => process.type === 'release')
+    if (releasePhase) {
+      ux.action.start('Checking release phase')
+      const {body: releases} = await this.heroku.get<Heroku.Release[]>(`/apps/${app}/releases`, {
+        partial: true,
+        headers: {
+          Range: 'version ..; max=5, order=desc',
+        },
+      })
+
+      const attach = releases.find(release => release.description?.includes('Attach DATABASE'))
+      const detach = releases.find(release => release.description?.includes('Detach DATABASE'))
+      if (!attach || !detach) {
+        ux.error('Unable to check release phase. Check your Attach DATABASE release for failures.')
+      }
+
+      const endTime = Date.now() + 900000 // 15 minutes from now
+      const [attachId, detachId] = [attach?.id as string, detach?.id as string]
+      while (true) {
+        const attach = await getRelease(this.heroku, app, attachId)
+        if (attach && attach.status === 'succeeded') {
+          let msg = 'pg:promote succeeded.'
+          const detach = await getRelease(this.heroku, app, detachId)
+          if (detach && detach.status === 'failed') {
+            msg += ` It is safe to ignore the failed ${detach.description} release.`
+          }
+
+          return ux.action.stop(msg)
```

Comment:
`return`'s like this one are correct because `ux.action.stop` also returns `void`. But written in this way it seems like there's some useful value being returned by `ux.action.stop` that will become the return value for `run()`. Maybe:

 ```suggestion
          ux.action.stop(msg)
          return
```